identity-service/src/solana-client.js

```diff
@@ -238,150 +232,10 @@ async function createTrackListenTransaction({
-  ;(async () => {
-    // eslint-disable-next-line no-unmodified-loop-condition
-    while (!done && getUnixTs() - startTime < timeout) {
-      connection.sendRawTransaction(rawTransaction, {
-        skipPreflight: true,
-        maxRetries: 0
-      })
-      await delay(300)
-    }
-    const elapsed = getUnixTs() - startTime
-    logger.info(
-      `TrackListen | Exited retry send loop for ${txid}, elapsed=${elapsed}, done=${done}, timeout=${timeout}, startTime=${startTime}`
-    )
-  })()
-
-  try {
-    await awaitTransactionSignatureConfirmation(
-      txid,
-      timeout,
-      connection,
-      logger
-    )
-  } catch (e) {
-    throw new Error(e)
-  } finally {
-    done = true
-  }
-  return txid
-}
-
-// Adapted from mango send function
-// https://github.com/blockworks-foundation/mango-ui/blob/b6abfc6c13b71fc17ebbe766f50b8215fa1ec54f/src/utils/send.tsx#L785
-// THIS FUNCTION MUST BE MOVED TO LIBS TRANSACTIONHANDLER
-async function awaitTransactionSignatureConfirmation(
-  txid,
-  timeout,
-  connection,
-  logger
-) {
-  let done = false
-  const result = await new Promise((resolve, reject) => {
-    ;(async () => {
-      setTimeout(() => {
-        if (done) {
-          return
-        }
-        done = true
-        reject(new Error(`Timed out for txid ${txid}`))
-      }, timeout)
-      try {
-        connection.onSignature(
-          txid,
-          (result) => {
-            done = true
-            if (result.err) {
-              reject(result.err)
-            } else {
-              resolve(result)
-            }
-          },
-          connection.commitment
-        )
-      } catch (e) {
-        done = true
-        logger.error('TrackListen | WS error in setup', txid, e)
-      }
-      while (!done) {
-        // eslint-disable-next-line no-loop-func
-        ;(async () => {
-          try {
-            const signatureStatuses = await connection.getSignatureStatuses([
-              txid
-            ])
-            const result = signatureStatuses && signatureStatuses.value[0]
-            if (!done) {
-              if (!result) {
-              } else if (result.err) {
-                logger.error('TrackListen | REST error for', txid, result)
-                done = true
-                reject(result.err)
-              } else if (
-                !(
-                  result.confirmations ||
-                  result.confirmationStatus === 'confirmed' ||
-                  result.confirmationStatus === 'finalized'
-                )
-              ) {
-              } else {
-                done = true
-                resolve(result)
-              }
-            }
-          } catch (e) {
-            if (!done) {
-              logger.error('REST connection error: txid', txid, e)
-            }
-          }
-        })()
-        await delay(300)
-      }
-    })()
-  })
-  done = true
-  return result
+  return [secpInstruction, listenInstruction]
```

Comment:
need to return the instructions instead of the tx for use in TransactionHandler

Reply:
seems like a design flaw of `TransactionHandler` tbh